x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/send_telemetry_events.ts

```diff
@@ -70,7 +71,7 @@ export function sendAlertTelemetryEvents(
     selectedEvents = enrichEndpointAlertsSignalID(selectedEvents, signalIdMap);
   }
   try {
-    eventsTelemetry.queueTelemetryEvents(selectedEvents);
+    eventsTelemetry.sendAsync(TelemetryChannel.ENDPOINT_ALERTS, selectedEvents);
```

Comment:
I believe `selectedEvents` here would be _all_ alerts (detection and endpoint), so I don't think this channel choice is totally accurate. Is separating those types (for telemetry) a goal of this PR?

Reply:
Good catch! thanks

Reply:
Yes, it makes sense, but since [the previous method](https://github.com/szaffarano/kibana/blob/szaffarano/telemetry-sender-v2/x-pack/plugins/security_solution/server/lib/telemetry/sender.ts#L318), `queueTelemetryEvents,`, always sends the events to the `alerts-endpoint` channel, I kept this behavior.
Do you think we should split `selectedEvents` so that we have two lists: for detection and endpoint events?

Reply:
@rylnd regarding your question, I realized that `selectEvents` filters out non  endpoint alerts events: L30: `sources.filter((obj: TelemetryEvent) => obj.data_stream?.dataset === 'endpoint.alerts')`, so I think it's fine to sent all `selectedEvents` to `TelemetryChannel.ENDPOINT_ALERTS`
WDYT?